cloud-aws-common/src/test/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsPlatformResourcesTest.java

```diff
@@ -317,6 +319,40 @@ public void collectEncryptionKeysWhenWeGetBackInfoThenItShouldReturnListWithElem
         assertEquals(4L, cloudEncryptionKeys.getCloudEncryptionKeys().size());
     }
 
+    @Test
+    public void collectEncryptionKeysWhenWeGetBackInfoThenItShouldReturnListWithPaginatedElements() {
```

Comment:
I cannot see any form of pagination being used here. Neither `describeKeyResult` nor `describeAliasResult` seems to ever return a non-null `nextMarker`.

Reply:
When the KMS_KEYS_COUNT is greater than 50(In this case I am creating 100 keys) , the results from describeKeyResult or describeAliasResult are paginated. So the goal here is that the result should contain all the 100 keys.  Please correct me If I have got your question wrong here.

Reply:
> When the KMS_KEYS_COUNT is greater than 50(In this case I am creating 100 keys) , the results from describeKeyResult or describeAliasResult are paginated.

No, they are not paginated automatically. The pagination logic is implemented in KMS that is replaced with a mock implementation here. `getListKeyResult()` and `getListAliasResult()` both rely on the value of `nextMarker` to decide whether any further elements remain to be described, keeping polling KMS until `nextMarker` becomes `null`. The test case shall simulate this by returning at least two responses for `client.listKeys()` (and similarly for `client.listAliases()`), where the all but the least response contains a non-null `nextMarker` and the last one has `nextMarker == null`.

Your current test case logic simply returns a single response containing 100 elements. No pagination is involved here at all, `nextMarker` is already `null` in both `describeKeyResult` and `describeAliasResult`.

Reply:
Sorry, correction: `describeKeyResult` is surely not a paged thing; we are always describing a single key in `AwsPlatformResources.encryptionKeys()`. What I rather meant was that `listKeysResult` and `describeAliasResult` should be paged. So the mock stubbing of `awskmsClient.listKeys()` and `awskmsClient.listAliases()` shall return at least two objects: the initial ones all using `nextMarker != null` and the last one having `nextMarker == null`. That way `getListKeyResult()` and `getListAliasResult()` will both be forced to call `awskmsClient` multiple times to fetch all pages of the resultset.